### ISSUE TITLE
BehaviorSubject (in Ionic Project) behaves strangely after update to latest Angular (v6 to v7)

### ISSUE THREAD

**Issue by mastix** [ionic-team/ionic#17904]
Hi guys,

I've checked out a very simple [ionic v4 Project with a login flow](https://github.com/meumobi/meu-starter.login-flow.ionic-v4).

The demo worked perfectly fine, but since this project is already 5 months old, I've decided to create a new ionic project with the latest ionic cli and put the source into the new one... with the latest stable ionic dependencies.

Besides other updates, the update of the ionic project updated the angular version from v6.1 to v7.2 (including its deps) and TypeScript from 2.9 to 3.1.

RXJS has been updated from 6.2.2 to 6.3.3.

But then the rxJS-BehaviorSubject (here: authState$) acts totally weird, see following screenshot.

Code: https://github.com/meumobi/meu-starter.login-flow.ionic-v4/blob/master/src/app/core/auth/auth.service.ts

When using the BehaviorSubject with the latest Ionic dependencies, it acts strangely:

Code:

```
console.log('isAuthenticated: ', this.authState$.getValue());
console.log('isAuthenticated: ', this.authState$);
```

Behaviour in console:

<img width="440" alt="Screenshot 2019-03-28 at 12 06 16" src="https://user-images.githubusercontent.com/1812978/55154014-21578f80-5154-11e9-88fb-a95616d807cc.png">

So what happens:

The getValue() returns the initial value, where as the printed object contains the correct value (true). This seems strange to me as they're both called right after each other.

I would be really cool if you could tell me what's going on.

Thanks so much in advance,

Sascha

P.S.: Stackoverflow ticket created in parallel - with all necessary information: https://stackoverflow.com/q/55396235/344514

**Ionic version:**
[x] **4.x**

**Ionic info:**
```
Ionic:

   ionic (Ionic CLI)             : 4.12.0 (/usr/local/lib/node_modules/ionic)
   Ionic Framework               : @ionic/angular 4.1.2
   @angular-devkit/build-angular : 0.13.7
   @angular-devkit/schematics    : 7.2.4
   @angular/cli                  : 7.3.7
   @ionic/angular-toolkit        : 1.4.1

Cordova:

   cordova (Cordova CLI) : not installed
   Cordova Platforms     : not available
   Cordova Plugins       : not available

System:

   Android SDK Tools : 26.1.1 (/usr/local/share/android-sdk)
   NodeJS            : v8.15.1 (/usr/local/Cellar/node@8/8.15.1/bin/node)
   npm               : 6.4.1
   OS                : macOS Mojave
   Xcode             : Xcode 10.2 Build version 10E125

**Comment by mastix** [comment]
When I change back the versions (to the original package.json) it will work again.

<img width="902" alt="Screenshot 2019-03-28 at 13 00 54" src="https://user-images.githubusercontent.com/1812978/55156198-94afd000-5159-11e9-8f1f-57bc5d14bc72.png">

**Comment by mastix** [comment]
I might have tracked it down to one single dependency that causes this issue.

Last one working: "@ionic/angular": "4.0.0-beta.17"
Not working: anything higher than @ionic/angular@4.0.0-beta.17.

When creating a new ionic project, it ships with 4.1.0. Which means that it cannot work with a new project. I can pretty much test it by changing the version... from beta.17 to beta.18 or the latest version.

**Comment by mhartington** [comment]
Hi there! This issue does not appear to be an issue with the framework itself, so it will be closed. Please review your issue and open a new ticket on the appropriate project repository, or feel free to post a question on the [Ionic Forum](http://forum.ionicframework.com). Thanks!
